internal/prompt/editor.go

```diff
@@ -1,68 +1,86 @@
+	cliEditors = []string{"vi", "vim", "nano"}
+)
 
-var defaultEditorPrompt = &editorPrompt{defaultEditor: defaultEditor}
+var defaultEditorPrompt = &editorPrompt{editorCmd: getDefaultEditor()}
 
 // CaptureInputViaEditor is the high level function to use in this package in
 // order to capture input from an editor.
 //
 // The arguments have been tailored for our use of strings mostly in the rest
 // of the CLI even though internally we're using []byte.
-func CaptureInputViaEditor(contents, pattern string) (result string, err error) {
-	v, err := defaultEditorPrompt.captureInput([]byte(contents), pattern)
+func CaptureInputViaEditor(contents, pattern string, infoFn func()) (result string, err error) {
+	v, err := defaultEditorPrompt.captureInput([]byte(contents), pattern, infoFn)
 	return string(v), err
 }
 
 type editorPrompt struct {
-	defaultEditor string
+	editorCmd string
 }
 
-// GetPreferredEditorFromEnvironment returns the user's editor as defined by the
-// `$EDITOR` environment variable, or the `defaultEditor` if it is not set.
-func (p *editorPrompt) getPreferredEditor() string {
-	editor := os.Getenv("EDITOR")
-
-	if editor == "" {
-		return p.defaultEditor
+// getDefaultEditor is taken from https://github.com/cli/cli/blob/trunk/pkg/surveyext/editor_manual.go
+// and tries to infer the editor from different heuristics.
+func getDefaultEditor() string {
+	if runtime.GOOS == "windows" {
+		return "notepad"
+	} else if g := os.Getenv("GIT_EDITOR"); g != "" {
+		return g
+	} else if v := os.Getenv("VISUAL"); v != "" {
+		return v
+	} else if e := os.Getenv("EDITOR"); e != "" {
+		return e
 	}
 
-	return editor
+	return defaultEditor
 }
 
-func (p *editorPrompt) resolveEditorArguments(executable string, filename string) []string {
-	args := []string{filename}
-
-	if strings.Contains(executable, "Visual Studio Code.app") {
-		args = append([]string{"--wait"}, args...)
+// openFile opens filename in the preferred text editor, resolving the
+// arguments with editor specific logic.
+func (p *editorPrompt) openFile(filename string, infoFn func()) error {
+	args, err := shellquote.Split(p.editorCmd)
+	if err != nil {
+		return err
 	}
+	args = append(args, filename)
 
-	// TODO(cyx): add other common editors
-
-	return args
-}
+	isCLIEditor := false
+	for _, e := range cliEditors {
+		if e == args[0] {
+			isCLIEditor = true
+		}
+	}
 
-// openFile opens filename in the preferred text editor, resolving the
-// arguments with editor specific logic.
-func (p *editorPrompt) openFile(filename string) error {
-	// Get the full executable path for the editor.
-	executable, err := exec.LookPath(p.getPreferredEditor())
+	editorExe, err := exec.LookPath(args[0])
 	if err != nil {
 		return err
 	}
 
-	cmd := exec.Command(executable, p.resolveEditorArguments(executable, filename)...)
+	cmd := exec.Command(editorExe, args[1:]...)
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
+	if !isCLIEditor && infoFn != nil {
```

Comment:
What happens if we show the message anyway? Would that cause any problems? Because if not, even if the message won't be read, we won't have to maintain a hardcoded list of editors.

Reply:
IMO it's just a tad weird if we do show it. I'm mostly "ok" with the idea of maintaining the list given it's such a short list of mainstream text based editors that people use anyway.

Here's a demo showing it https://asciinema.org/a/K7wMo0sBRWtq1e1qMtyJhAiZ2

It's likely just my OCD coming into play :)

Reply:
Ok, given that the list won't change much. We can add `emacs` and `neovim` to it, and it will probably stay unchanged for a long time.